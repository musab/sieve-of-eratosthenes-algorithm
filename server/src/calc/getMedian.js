export const getMedian = nums => {
  if (!nums.length || !Array.isArray(nums)) {
    throw new Error('Must supply an Array of Numbers');
  }
  const numsLen = nums.length;
  let median = 0;

  // determine if length of nums is even or odd
  // if even return more left and right of median
  if (numsLen % 2 === 0) {
    median = [nums[numsLen / 2 - 1], nums[numsLen / 2]];
  } else {
    median = [nums[(numsLen - 1) / 2]];
  }

  return median;
};
