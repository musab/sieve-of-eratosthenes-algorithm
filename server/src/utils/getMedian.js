export const getMedian = nums => {
  const numsLen = nums.length;
  let median = 0;

  // determine if length of nums is even or odd
  if (numsLen % 2 === 0) {
    median = [nums[numsLen / 2 - 1], nums[numsLen / 2]];
  } else {
    median = nums[(numsLen - 1) / 2];
  }

  return median;
};
