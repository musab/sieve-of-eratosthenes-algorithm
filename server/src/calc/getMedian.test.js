import { getMedian } from './getMedian';

describe('getMedian', () => {
  test('[2, 3, 5, 7] returns 3, 5', () => {
    const input = [2, 3, 5, 7];
    const expected = [3, 5];

    expect(getMedian(input)).toStrictEqual(expected);
  });

  test('[2, 3, 5, 7, 11, 13, 17] returns 7', () => {
    const input = [2, 3, 5, 7, 11, 13, 17];
    const expected = [7];

    expect(getMedian(input)).toStrictEqual(expected);
  });

  test('[0] returns 0', () => {
    const input = [0];
    const expected = [0];

    expect(getMedian(input)).toStrictEqual(expected);
  });

  test('if called with non array type throws error', () => {
    const input = '1';
    function passNonArray() {
      getMedian(input);
    }

    expect(passNonArray).toThrowErrorMatchingSnapshot();
  });

  test('if called with empty array throws error', () => {
    const input = [];
    function passEmptyArr() {
      getMedian(input);
    }

    expect(passEmptyArr).toThrowErrorMatchingSnapshot();
  });
});
