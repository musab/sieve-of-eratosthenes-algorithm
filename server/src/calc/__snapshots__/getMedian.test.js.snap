// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getMedian if called with empty array throws error 1`] = `"Must supply an Array of Numbers"`;

exports[`getMedian if called with non array type throws error 1`] = `"Must supply an Array of Numbers"`;
